src/main/java/org/opentripplanner/routing/algorithm/filterchain/groupids/GroupByUtils.java

```diff
@@ -0,0 +1,18 @@
+package org.opentripplanner.routing.algorithm.filterchain.groupids;
+
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.site.StationElement;
+import org.opentripplanner.transit.model.site.StopLocation;
+
+public class GroupByUtils {
+
+  /**
+   * Get the parent station id if such exists. Otherwise, return the stop id.
+   */
+  protected static FeedScopedId getStopOrStationId(StopLocation stopPlace) {
+    if (stopPlace instanceof StationElement stationElement && stationElement.isPartOfStation()) {
+      return stationElement.getParentStation().getId();
+    }
+    return stopPlace.getId();
+  }
```

Comment:
This method belong in the StopLocation and `getStationOrStopId` is a better name - you get station-id if it exist, then stop is backup.

Reply:
Moved to StopLocation.